### PR TITLE
Polars writing

### PR DESCRIPTION
**Pull request by stephenpardy**
closes: #433 

---

## What
  * Adds support for logging polars dataframes

## How to Test
  * Run the unit tests

### CODE REVIEW

File: environment.yml [hunk 1]
```diff
@@ -39,6 +39,7 @@ dependencies:
   - pytest
   - pytest-cov
   - xgboost
+  - polars<1.0
```
Comment (ryanSoley) [inline environment.yml:42]: any particular reason for the pin?

File: rubicon_ml/client/mixin.py [hunk 2]
```diff
@@ -23,6 +23,7 @@
 if TYPE_CHECKING:
     import dask.dataframe as dd
     import pandas as pd
+    from polars import DataFrame as polars_DataFrame
```
Comment (ryanSoley) [inline rubicon_ml/client/mixin.py:26]: nit: can we do the import consistently? either like this or with `import polars as pl` like the rest of the files

File: rubicon_ml/repository/base.py [hunk 3]
```diff
@@ -602,9 +608,14 @@ def _persist_dataframe(self, df, path):
...
-        df.to_parquet(path, engine="pyarrow")
+        if hasattr(df, "write_parquet"):
+            # handle Polars
+            df.write_parquet(path)
+        else:
+            # Dask or pandas
+            df.to_parquet(path, engine="pyarrow")
 
-    def _read_dataframe(self, path, df_type="pandas"):
+    def _read_dataframe(self, path, df_type: Literal["pandas", "dask", "polars"] = "pandas"):
```
Comment (ryanSoley) [inline rubicon_ml/repository/base.py:618]: this function needs to be updated to read polars still, right?

File: tests/unit/repository/test_base_repo.py [hunk 4]
```diff
@@ -488,6 +523,23 @@ def test_create_dask_dataframe(memory_repository):
...
+
+    open_file = repository.filesystem.open(dataframe_metadata_path)
+    with open_file as f:
+        dataframe_json = json.load(f)
+
+    assert repository.filesystem.exists(dataframe_data_path)
+    assert dataframe.id == dataframe_json["id"]
+
+
 def test_get_pandas_dataframe(memory_repository):
```
Comment (ryanSoley) [inline tests/unit/repository/test_base_repo.py:543]: we need a getter test similar to this one for polars

File: environment.yml (hunk 1 above)
Comment (stephenpardy) [reply environment.yml:42]: New version with some breaking changes just released. https://docs.pola.rs/releases/upgrade/1/ Would like to wait to double check before allowing this new version.

File: rubicon_ml/repository/base.py [hunk 5]
```diff
@@ -709,7 +737,13 @@ def get_dataframes_metadata(
 
         return self._load_metadata_files(dataframe_metadata_root, domain.Dataframe)
 
-    def get_dataframe_data(self, project_name, dataframe_id, experiment_id=None, df_type="pandas"):
+    def get_dataframe_data(
+        self,
+        project_name: str,
+        dataframe_id: str,
+        experiment_id: Optional[str] = None,
+        df_type: Literal["pandas", "dask", "polars"] = "pandas",
```
Comment (ryanSoley) [inline rubicon_ml/repository/base.py:745]: we can probably just infer this and store it in the metadata so users don't have to keep specifying

**Review by ryanSoley — APPROVED** [review]
looks great! thanks!

**Pull request merged**